rotkehlchen/chain/ethereum/modules/defisaver/decoder.py

```diff
@@ -10,18 +11,98 @@
 )
 from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
 from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.globaldb.handler import GlobalDBHandler
 from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
 from rotkehlchen.logging import RotkehlchenLogsAdapter
-from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
 from rotkehlchen.utils.misc import bytes_to_address
 
-from .constants import CPT_DEFISAVER, DEACTIVATE_SUB, SUB_STORAGE, SUBSCRIBE
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+from .constants import (
+    CPT_DEFISAVER,
+    DEACTIVATE_SUB,
+    DFS_GENERIC_LOGGER_LEGACY,
+    FL_AAVEV2,
+    FL_AAVEV3,
+    FL_AAVEV3_CARRY_DEBT,
+    FL_AAVEV3_WITH_FEE,
+    FL_ACTION_V1_0_0,
+    FL_ACTION_V1_0_1,
+    FL_ACTION_V1_0_2,
+    FL_ACTION_V1_0_3,
+    FL_ACTION_V1_0_3_BIS,
+    FL_BALANCER,
+    FL_BALANCER_VPREV_1,
+    FL_BALANCER_VPREV_2,
+    FL_DYDX,
+    FL_DYDX_VPREV_1,
+    FL_DYDX_VPREV_2,
+    FL_EULER_V1_0_0,
+    FL_EULER_V1_0_1,
+    FL_GHO,
+    FL_MAKER_V1_0_0,
+    FL_MAKER_V1_0_1,
+    FL_MAKER_V1_0_2,
+    FL_MAKER_V1_0_3,
+    FL_MORPHO_BLUE,
+    FL_SPARK,
+    FL_UNIV3,
+    SUB_STORAGE,
+    SUBSCRIBE,
+)
 
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
 class DefisaverDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.flashloan_addresses = (
+            FL_ACTION_V1_0_0,
+            FL_ACTION_V1_0_1,
+            FL_ACTION_V1_0_2,
+            FL_ACTION_V1_0_3,
+            FL_ACTION_V1_0_3_BIS,
+            FL_MORPHO_BLUE,
+            FL_BALANCER,
+            FL_BALANCER_VPREV_1,
+            FL_BALANCER_VPREV_2,
+            FL_AAVEV3,
+            FL_EULER_V1_0_1,
+            FL_EULER_V1_0_0,
+            FL_SPARK,
+            FL_AAVEV3_WITH_FEE,
+            FL_AAVEV3_CARRY_DEBT,
+            FL_AAVEV2,
+            FL_DYDX,
+            FL_DYDX_VPREV_1,
+            FL_DYDX_VPREV_2,
+            FL_GHO,
+            FL_MAKER_V1_0_3,
+            FL_MAKER_V1_0_2,
+            FL_MAKER_V1_0_1,
+            FL_MAKER_V1_0_0,
+            FL_UNIV3,
+        )
+        self.dfs_logger_addresses = (
+            DFS_GENERIC_LOGGER_LEGACY,
+        )
```

Comment:
```suggestion
        self.dfs_logger_addresses = DFS_GENERIC_LOGGER_LEGACY
```

Reply:
For now I am keeping it as a tuple ready for adding other logger addresses